source/Calamari.AzureAppService/ArmClientExtensions.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure;
+using Azure.ResourceManager;
+using Azure.ResourceManager.AppService;
+using Azure.ResourceManager.AppService.Models;
+using Calamari.AzureAppService.Azure;
+using Calamari.AzureAppService.Json;
+using Octopus.CoreUtilities.Extensions;
+
+namespace Calamari.AzureAppService
+{
+    ///<summary>
+    /// Provides a set of static methods for interacting with an <see cref="ArmClient"/> using an <see cref="AzureTargetSite"/>.
+    ///</summary>
+    /// <remarks>
+    /// These methods are suffixed with <i>Async</i> for consistency with the <see cref="ArmClient"/>.
+    /// In the <b>Azure.ResourceManager</b> SDKs, <i>Async</i>-suffixed methods indicate an API call is made to Azure.
+    /// </remarks>
+    public static class ArmClientExtensions
+    {
+        public static async Task<SiteConfigData> GetSiteConfigDataAsync(this ArmClient armClient, AzureTargetSite targetSite)
+        {
+            return targetSite.HasSlot switch
+                   {
+                       true => (await armClient.GetWebSiteSlotConfigResource(WebSiteSlotConfigResource.CreateResourceIdentifier(
```

Comment:
nit: I have no strong opinion on this one, but it feels weird to switch on boolean value. I think if statements are a more natural way. But this isn't blocking, just a style choice.

Reply:
Yeah, I can understand that. I like switch statements like this because I find them a bit easier to read.